### ISSUE TITLE
Flow questions for sidebars

### ISSUE THREAD

**Issue by schalkneethling** [mozilla/all-aboard#1]
@chrismore Couple of questions about the flow here:

**Scenario**
User installs and opens Fx, the user answers the questions and either completes the FxA form or simply navigates away from /firstrun.

Do we store the time it was shown as the start time for our first 24 hour interval or, do we wait until the user closes the sidebar, and only then store the start time for the interval?

After 24 hours, we then update the add-on icon with a badge and trigger the transient notification. Now, do we start our next interval as soon as the user clicks either the notification or the icon or, do we only start the next interval once the user has interacted with the CTA of the sidebar?

**Comment by chrismore** [comment]
> Do we store the time it was shown as the start time for our first 24 hour interval or, do we wait until the user closes the sidebar, and only then store the start time for the interval?

I would say we should start the timer as soon as the bar first opens after the questions. If the bar is never closed and they keep it open, will the user see the content change at 24 hours later without even clicking anything?

> After 24 hours, we then update the add-on icon with a badge and trigger the transient notification. Now, do we start our next interval as soon as the user clicks either the notification or the icon or, do we only start the next interval once the user has interacted with the CTA of the sidebar?

As for future sidebars and intervals, there are two options:

1) the timer starts for counting the next interval *only* if they have opened the sidebar or clicked the notification and/or button. This will prevent a user from skipping ahead in onboarding if they don't interact with a sidebar. Also, if the don't interact with a sidebar, we won't keep pinging them 5 times for future content. This will also keep them from getting to the final sticker promotion without ever doing anything.

OR

2) The timer starts for the next content sidebar as soon as we trigger a notification. The issue with this is that users could skip ahead by not engaging with a sidebar and we could ping them many times for future content even if they haven't engaged with anything. 

I think I am leaning toward #1.

**Comment by chrismore** [comment]
> I guess following on from this is updating the distribution.id Do we want to update it only once or, do we want to do two things

Let's just append what step they are on and that is simply what content is being presented to them now. It doesn't matter if they never open it or never clicked anything. Just what step they are at. Trying not to get too close to behaviors and just the general flow.

**Comment by schalkneethling** [comment]
> I would say we should start the timer as soon as the bar first opens after the questions. If the bar is never closed and they keep it open, will the user see the content change at 24 hours later without even clicking anything?

Yes, it should simply update. Will have to test this but, that is how the API is supposed to work.

I agree, let's go with option 1 for future sidebars. With that said, a badge is only given if the CTA has actually been clicked correct?

**Comment by chrismore** [comment]
> I agree, let's go with option 1 for future sidebars.

Option 1 it is!

> With that said, a badge is only given if the CTA has actually been clicked correct?

We are going to make it even easier. The badge will just be there by simply opening up the sidebar. The badges just become more of a progress bar. The reward is simply just opening up the bar. The badges can just be there in the sidebar already as the template for that content number (1-5). No badge is given for the import data sidebar and all 5 badges are blank on the first import data sidebar.

**Comment by schalkneethling** [comment]
Ok, so we are decided on the following:

The timer starts counting the next interval once the user have opened the sidebar from either the add-on icon or notification. This will prevent a user from skipping ahead in onboarding if they don't interact with a sidebar. Also, if they don't interact with a sidebar, we won't keep pinging them 5 times for future content. This will also keep them from getting to the final sticker promotion without ever doing anything.

The `distribution.id` will simply indicate what step they were on.

A badge is given simply for opening up a sidebar.

Agreed? @chrismore @FrancescoSTL

**Comment by fabiorios** [comment]
Nice, that's what I understood.

**Comment by chrismore** [comment]
> The timer starts counting the next interval once the user have opened the sidebar from either the add-on icon or notification. This will prevent a user from skipping ahead in onboarding if they don't interact with a sidebar. Also, if they don't interact with a sidebar, we won't keep pinging them 5 times for future content. This will also keep them from getting to the final sticker promotion without ever doing anything.

Makes sense to me and how I think makes most sense for the UX.

**Comment by schalkneethling** [comment]
@chrismore Have we decided to show the token/badge after the user has taken action or, is it still as defined in comment #5 above?

**Comment by fabiorios** [comment]
@schalkneethling Same as comment #5. Just open the panel, no other action needed to get a badge.

**Comment by schalkneethling** [comment]
Closing this as the logic has been merged into master.
